cmd/snap/cmd_buy.go

```diff
@@ -110,26 +110,43 @@ func buySnap(opts *store.BuyOptions) error {
 	if !paymentInfo.AllowsAutomaticPayment {
 		w := tabwriter.NewWriter(Stdout, 0, 3, 2, ' ', 0)
 
+		index := -1
+
 		fmt.Fprintln(w, i18n.G("\tSelection\tDescription"))
 		for i, method := range paymentInfo.Methods {
-			fmt.Fprintf(w, "\t%d\t%s\n", i+1, method.Description)
+			preferred := ""
+			if method.Preferred {
+				preferred = "*"
+				index = i + 1
+			}
+			fmt.Fprintf(w, "%s\t%d\t%s\n", preferred, i+1, method.Description)
```

Comment:
here I would've defaulted `preferred` to a space, and used `%s%d``, so the output is more compact.